src/shapes.rs

```diff
@@ -74,6 +74,36 @@ pub fn draw_rectangle_lines(x: f32, y: f32, w: f32, h: f32, thickness: f32, colo
     context.gl.geometry(&vertices, &indices);
 }
 
+/// Draws a solid rectangle with its center at `position` with `size`,
+/// with rotation (in degrees), with a given `color`.
+pub fn draw_rectangle_pro(position: Vec2, rotation: f32, size: Vec2, color: Color) {
```

Comment:
for "optional arguments" macroquad use `do_whatever_ex` + a struct with arguments notation

you can check `draw_texture_ex` as an example